android/modules/calendar/src/java/ti/modules/titanium/calendar/EventProxy.java

```diff
@@ -243,6 +244,36 @@ public static EventProxy createEvent(CalendarProxy calendar, KrollDict data)
 		}
 	}
 
+	private AttendeeProxy[] getAttendeeProxies() {
+		final String[] attendeeProjection = new String[]{
+				CalendarContract.Attendees._ID,
+				CalendarContract.Attendees.EVENT_ID,
+				CalendarContract.Attendees.ATTENDEE_NAME,
+				CalendarContract.Attendees.ATTENDEE_EMAIL,
+				CalendarContract.Attendees.ATTENDEE_TYPE,
+				CalendarContract.Attendees.ATTENDEE_RELATIONSHIP,
+				CalendarContract.Attendees.ATTENDEE_STATUS
+		};
+		final String query = "(" + CalendarContract.Attendees.EVENT_ID + " = ?)";
+		final String[] args = new String[]{id};
+		ContentResolver contentResolver = TiApplication.getInstance().getContentResolver();
+		final Cursor cursor = contentResolver.query(CalendarContract.Attendees.CONTENT_URI, attendeeProjection, query, args, null);
+		int index = 0;
+		AttendeeProxy[] result = new AttendeeProxy[cursor.getCount()];
+		while (cursor.moveToNext()) {
+			//create a proxy instance
+			AttendeeProxy proxyForRow = new AttendeeProxy(cursor.getString(cursor.getColumnIndex(CalendarContract.Attendees.ATTENDEE_EMAIL)),
+															cursor.getString(cursor.getColumnIndex(CalendarContract.Attendees.ATTENDEE_NAME)),
+															cursor.getInt(cursor.getColumnIndex(CalendarContract.Attendees.ATTENDEE_TYPE)),
+															cursor.getInt(cursor.getColumnIndex(CalendarContract.Attendees.ATTENDEE_STATUS)),
+															cursor.getInt(cursor.getColumnIndex(CalendarContract.Attendees.ATTENDEE_RELATIONSHIP))
```

Comment:
The `cursor.getColumnIndex()` call can return `-1` which would trigger an out-of-bounds exception when passed through to the cursor getter methods. Also, the cursor getter methods can throw an exception if the column type does not match the type requested (ie: it stores a string when we're requesting an int).

To be safe, we should make sure the column index is valid before using it. And wrap each cursor getter in separate try/catch blocks. We can default fields not found to constants such as NONE.

Also, should we define our own "UNKNOWN" constant? The idea is to future-proof this code for when Google adds new constants that our internal code doesn't support yet.